chth0n1x/ember-osf-web
# [ENG-2841] Update Dropdown UI Changes

-   Ticket: https://openscience.atlassian.net/browse/ENG-2841
-   Feature flag: n/a

## Purpose

The purpose of this change is to improve the overall UI layout of the update dropdown including the trigger and the component. 

## Summary of Changes

- Added flex CSS logic to the trigger  for rows
- Added flex CSS logic to the list elements for columns
- Added width element for the Registries states for improved formatting
- Adding padding to the top and left of the Registries States for improved formatting

## Screenshot(s)

Additional updates to follow for the caret-down and padding for list, but current state:
![image](https://user-images.githubusercontent.com/82047646/142044558-e2a9350b-545d-4bfa-b75a-e119f81efbc5.png)

![image](https://user-images.githubusercontent.com/82047646/142044658-34cf69c4-89f2-4edb-aa35-4dc1067dea3b.png)

![image](https://user-images.githubusercontent.com/82047646/142044728-0a11c3b7-7f97-4531-b10e-540aaa2f3d32.png)

![image](https://user-images.githubusercontent.com/82047646/142044782-44f85443-8d43-4a79-bec2-40be8aef2f90.png)


## Side Effects

The dropdown has the potential to display many changes, and how it obscures the main view can have consequences on the overall UX. Positioning has been added to the overview-topbar components to ensure it does not affect the registries side nav component.

## QA Notes

Testing should occur for null array conditions, one approved edit, one in progress edit, and many (<20, >10) to see how the scroll functions once added. Other edge cases may be considered, but positioning, open/close functionality, and tests for focused elements and hover should be tested for optimal user experience.
